distage/distage-core-api/src/main/scala/izumi/distage/model/effect/QuasiIO.scala

```diff
@@ -51,6 +51,8 @@ trait QuasiIO[F[_]] extends QuasiApplicative[F] {
     */
   def definitelyRecoverCause[A](action: => F[A])(recoverCause: (Throwable, (() => Throwable)) => F[A]): F[A]
 
+  def redeem[A, B](action: F[A])(failure: Throwable => F[B], success: A => F[B]): F[B]
```

Comment:
@VladPodilnyk Are you sure a separate `redeem` method is required when it's equivalent to `action.definitelyRecover(failure).flatMap(success)`?

Reply:
```suggestion
  def redeem[A, B](action: => F[A])(failure: Throwable => F[B], success: A => F[B]): F[B]
```
Also you would need suspension here to make it work with `Identity`.

Reply:
>@VladPodilnyk Are you sure a separate redeem method is required when it's equivalent to action.definitelyRecover(failure).flatMap(success)?

The problem is action has type `F[A]`, failure is a function from `Throwable to F[B]` hence I couldn't use `definatelyRecover` since it requires the same type for action and failure. So I decided to add some functionality.

Reply:
Right, I wasn't correct here